packages/backend/src/managers/catalogManager.ts

```diff
@@ -93,4 +104,38 @@ export class CatalogManager extends Publisher<ApplicationCatalog> implements Dis
     }
     return recipe;
   }
+
+  async addLocalModelsToCatalog(models: LocalModelImportInfo[]): Promise<void> {
+    // we copy the current catalog in another object and update it with the model
+    // then write it to the custom catalog path. If it exists it will be overwritten by default
+    const tmpCatalog: ApplicationCatalog = Object.assign({}, this.catalog);
+
+    for (const model of models) {
+      const statFile = await promises.stat(model.path);
+      tmpCatalog.models.push({
+        id: model.path,
+        name: model.name,
+        description: `Model imported from ${model.path}`,
+        hw: 'CPU',
+        file: {
+          path: path.dirname(model.path),
+          file: path.basename(model.path),
+        },
+        memory: statFile.size,
+      });
+    }
+
+    const customCatalog = path.resolve(this.appUserDirectory, 'catalog.json');
```

Comment:
I think we should have two catalog here. I am not in favor of overwriting the _native_ catalog, I feel it would be more appropriate to have some kind of `user-models.json` in the `appUserDirectory` what do you think ?

Reply:
From my point of view the operation that the user is doing here it is just updating the custom catalog.json file that we already support. It is just that we guide them. But if someone wants to add a new model manually we are already saying that they need to create a `catalog.json` file (copy/paste the `assets/ai.json`) inside the ai-lab folder and update it. 

So i wouldn't create a new file as it may overcomplicate stuff, but let's see what others think

Reply:
How do you handle the case where we update on our side the catalog ? Are we losing the user imported models ?

Reply:
When importing, if the `catalog.json` file already exists, it just adds the new models in there and update the file on disk. If it does not exists it creates it on disk by copying the content from the native `ai.json` and adding the imported models. 

So if the user wants to update the catalog after having imported models he would work as before. Open the `catalog.json` file which would also contain the imported models and edit it.

Reply:
What if some day we want to remove a model ? like we did with llama2 ? the mechanism you describe would not be suitable ?

Reply:
This is the same thing that may happen today. If i create my custom catalog.json and we update ai.json, ai-lab will still point to the custom catalog. We have to refactor how the custom catalog + ai.json work or inform the user any update. Opening an issue

Reply:
-> https://github.com/containers/podman-desktop-extension-ai-lab/issues/792

Reply:
What about having the 2 files but merging them instead of choosing one over the other